### PR TITLE
Enable nolintlint linter

### PR DESCRIPTION
**Pull request by jakule**
`nolintlint` shows unused `nolint` directives and forces adding a comment about why the linter is disabled.

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -54,10 +55,15 @@ linters-settings:
       - prefix(github.com/gravitational/teleport) # Custom section: groups all imports with the specified Prefix.
     skip-generated: true # Skip generated files.
     custom-order: true # Required for "sections" to take effect.
+  nolintlint:
+    allow-unused: false
+    allow-no-explanation: [ ]
```
Comment (codingllama) [inline .golangci.yml:60]: nit: Remove default value?

```suggestion
```

File: .golangci.yml [hunk 2]
```diff
@@ -54,10 +55,15 @@ linters-settings:
...
+    require-explanation: true
+    require-specific: true
 output:
   uniq-by-line: false
 
 run:
   skip-dirs-use-default: false
   timeout: 5m
-  go: '1.18'
+  go: '1.19'
```
Comment (codingllama) [inline .golangci.yml:69]: Good, thanks.

File: lib/srv/desktop/rdp/rdpclient/client_common.go [hunk 3]
```diff
@@ -61,7 +61,7 @@ type Config struct {
 // GenerateUserCertFn generates user certificates for RDP authentication.
 type GenerateUserCertFn func(ctx context.Context, username string, ttl time.Duration) (certDER, keyDER []byte, err error)
 
-//nolint:unused
+//nolint:unused // useful for debugging?
```
Comment (codingllama) [inline lib/srv/desktop/rdp/rdpclient/client_common.go:64]: Is it really unused? Could we delete it or tag the author? It looks like someone wrote it but forgot to add the calls.

**Review by codingllama — APPROVED** [review]
Thanks!

File: lib/srv/desktop/rdp/rdpclient/client_common.go (hunk 3 above)
Comment (jakule) [reply lib/srv/desktop/rdp/rdpclient/client_common.go]: I've already fixed that. This function is used in a file that has a build flag.

**Review by zmb3 — APPROVED** [review]

File: .golangci.yml [hunk 4]
```diff
@@ -54,10 +55,14 @@ linters-settings:
       - prefix(github.com/gravitational/teleport) # Custom section: groups all imports with the specified Prefix.
     skip-generated: true # Skip generated files.
     custom-order: true # Required for "sections" to take effect.
+  nolintlint:
+    allow-unused: true
```
Comment (codingllama) [inline .golangci.yml:59]:
```suggestion
    allow-unused: true # Enabled because of conditional builds / build tags.
```

**Review by codingllama — APPROVED** [review]

**Pull request merged**